lib/todo.rb

```diff
@@ -46,5 +47,10 @@ def remove(item)
       #@TODO: Create remove functionality
     end #remove
 
+    def valid_removal?(response)
+      return false unless response.length == 1 && response.match?(/\d/) && response.to_i > 0 && response.to_i < get_contents.length + 1
+      return true
```

Comment:
That is a lot of conditions here.  This boils down to two condition cases:

- Do we have any items to remove
- Is the input within the item range

From the user's perspective, I think it would be more helpful to know there's nothing left on the list before being asked which item to remove.  If we separate that logic out, we're left with input validation.  With that, we can reduce this pretty substantially.

We know that unlike `Integer(input)`, calling `to_i` on a string containing a non-number input will result in a 0, which is not a valid number of this case.  Additionally, passing in `"0"` isn't valid because we're 1-indexing these items.  Thus, we can reduce our checks to whether the input is between 1 and the number of items in the list.  This can be expressed using [`Range`](https://ruby-doc.org/core-3.0.0/Range.html) and it's [`Range#member?`](https://ruby-doc.org/core-3.0.0/Range.html#method-i-member-3F) method.  Additionally, because this method returns a boolean, we can drop the return statement and the unless by using this.

```suggestion
      (1..contents.length+1).member?(response.to_i)
```

Reply:
I figured this had a lot of opportunity for refactoring.  Thanks for the help, my goal was to get it to work.